src/master_gateway.js

```diff
@@ -54,12 +54,18 @@ var masterGateway = {
     msgSender: 'Master',
 
     eventHandler: null,
-
+    
     init: function (config) {
         var slaves = config.slaves || config.products;
         this.initialized = true;
         this.config = config;
-        this.slaves = slaves || {};
+        if (slaves) {
+            for (var slave in slaves) {
+                if (slaves.hasOwnProperty(slave)) {
+                    this.addSlave(slaves[slave]);
+                }
+            }
+        }
```

Comment:
@thedev966 what is this doing, it doesn't seem like it is in task scope?

Reply:
@ZvonimirBusic Prior, there were two different ways to initialize slaves. So, Dinko and I concluded to stick with one. Regardless of whether `init() `or `addSlave()` directly are called, we should always go through `addSlave()` to register a slave.

Reply:
@thedev966 Is this breaking change, if app has `slaves` to init, will it behave different than before this change?